Content.Server/Chemistry/EntitySystems/ReagentDispenserSystem.cs

```diff
@@ -91,12 +92,11 @@ private List<string> GetInventory(ReagentDispenserComponent reagentDispenser)
 
         private void OnEmagged(EntityUid uid, ReagentDispenserComponent reagentDispenser, ref GotEmaggedEvent args)
         {
-            if (!reagentDispenser.IsEmagged)
-            {
-                reagentDispenser.IsEmagged = true;
-                args.Handled = true;
-                UpdateUiState(reagentDispenser);
-            }
+            // adding component manually to have correct state
+            args.Emag = false;
```

Comment:
Should the event raiser just use EnsureComp<T> instead.

Reply:
didnt know that was a thing, will do it in a bit